sjsonnet/src/sjsonnet/Std.scala

```diff
@@ -1281,6 +1203,49 @@ class Std {
       Platform.sha3(str)
     },
   )
+
+  private def validateSet(ev: EvalScope, pos: Position, keyF: Val, arr: Val): Unit = {
+    if (ev.settings.throwErrorForInvalidSets) {
+      val sorted = uniqArr(pos.noOffset, ev, sortArr(pos.noOffset, ev, arr, keyF), keyF)
+      if (!ev.equal(arr, sorted)) {
+        if (ev.settings.strictSetOperations) {
+          Error.fail(arr.prettyName + " is not a set")
+        } else {
+          System.err.println(arr.prettyName + " is not a set")
```

Comment:
Perhaps this should use use `ev.warn` instead of println`? `ev.warn` calls the `warnLogger` which is configured at higher layers.